### DISCUSSION TITLE
Logging Manager

### DISCUSSION THREAD

**Discussion by LukasVindbjerg** [isaac-sim/IsaacLab#353] [Ideas]
Hi, 

I'm curious about the current solution for logging data in orbit. I would like to log the different states of my robot during my experiments and use this data for plotting - but I'm not sure how to go about this. 

I was thinking of implementing a logging manager, that can take different functions and log the output of these to a file.

Before I go for this I just wanted to ask what is your current take on recording data from the simulation and saving it. I read the Recoding Animations section of the documentation, but I'm not sure this is what I want to do in my case.

**Comment by Mayankm96** [comment]
Hi,

We have been discussing this recently as well. One possible solution is:

* Add the quantities you want to log into its own ObservationGroup that is computed at every step
* Dump these into a file using data collector wrapper (similar to how we use them for imitation learning): https://github.com/NVIDIA-Omniverse/orbit/blob/main/source/standalone/workflows/robomimic/collect_demonstrations.py

Open to other suggestions as well :)
